File: aibom/tests/fixtures/simple_agent.ts
import { ChatOpenAI } from "@langchain/openai";
import { StateGraph } from "@langchain/langgraph";
import { MemorySaver } from "@langchain/langgraph";
import { ToolNode } from "@langchain/langgraph/prebuilt";
import { tool } from "@langchain/core/tools";

const llm = new ChatOpenAI({ model: "gpt-4o-mini" });

const searchTool = tool(async (input: string) => {
  return `Results for: ${input}`;
}, { name: "search", description: "Search the web" });

const graph = new StateGraph({});
graph.addNode("agent", llm);
graph.addNode("tools", new ToolNode([searchTool]));

const memory = new MemorySaver();
const app = graph.compile({ checkpointer: memory });

File: aibom/tests/fixtures/simple_agent.py
from langchain_openai import ChatOpenAI
from langgraph.graph import StateGraph
from langgraph.prebuilt import ToolNode
from langgraph.checkpoint.memory import MemorySaver
from langchain_core.tools import tool
from langchain_core.prompts import ChatPromptTemplate

llm = ChatOpenAI(model="gpt-4o-mini")

prompt = ChatPromptTemplate.from_messages([
    ("system", "You are a helpful assistant."),
    ("human", "{input}"),
])


@tool
def search(query: str) -> str:
    """Search the web."""
    return f"Results for: {query}"


graph = StateGraph(dict)
graph.add_node("agent", llm)
graph.add_node("tools", ToolNode([search]))
memory = MemorySaver()
app = graph.compile(checkpointer=memory)
